### PR TITLE
[NCL-5910] add DB migration script for artifact -built-by-> build rel…

### PR DESCRIPTION
**Pull request by janinko**
…ationship

### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql [hunk 1]
```diff
@@ -188,3 +188,34 @@ BEGIN transaction;
...
+BEGIN transaction;
+
+    -- Verify that there are no duplicate data
+    do $$
+    declare
+        total_dupes integer;
+    begin
+        SELECT count(*)
+        INTO total_dupes
+        FROM (SELECT built_artifact_id FROM build_record_built_artifact_map GROUP BY built_artifact_id HAVING count(*) > 1) AS agg;
```
Comment (michalovjan) [inline model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql:202]: What does "AS agg" accomplish? Is that intentional?

File: model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql [hunk 2]
```diff
@@ -188,3 +188,34 @@ BEGIN transaction;
...
+
+        assert total_dupes = 0;
+    end$$;
+
+    -- Add new column to artifact table
+    ALTER TABLE artifact ADD column buildrecord_id int4;
+    CREATE INDEX idx_artifact_buildrecord ON artifact (buildrecord_id);
+    ALTER TABLE artifact ADD CONSTRAINT fk_artifact_buildrecord FOREIGN KEY (fk_artifact_buildrecord) REFERENCES buildrecord(id);
```
Comment (michalovjan) [inline model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql:210]: It should be 
```suggestion
    ALTER TABLE artifact ADD CONSTRAINT fk_artifact_buildrecord FOREIGN KEY (buildrecord_id) REFERENCES buildrecord(id);
```

File: model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql [hunk 3]
```diff
@@ -188,3 +188,34 @@ BEGIN transaction;
...
+        SELECT count(*)
+        INTO total_dupes
+        FROM (SELECT built_artifact_id FROM build_record_built_artifact_map GROUP BY built_artifact_id HAVING count(*) > 1) AS agg;
+
+        assert total_dupes = 0;
+    end$$;
+
+    -- Add new column to artifact table
+    ALTER TABLE artifact ADD column buildrecord_id int4;
+    CREATE INDEX idx_artifact_buildrecord ON artifact (buildrecord_id);
```
Comment (michalovjan) [inline model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql:209]: Add the Index also to Artifact model here https://github.com/project-ncl/pnc/blob/b9b48674b743e92b36ff57960b820b0f3d26d2d7/model/src/main/java/org/jboss/pnc/model/Artifact.java#L71

File: model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql [hunk 4]
```diff
@@ -188,3 +188,34 @@ BEGIN transaction;
...
+    ALTER TABLE artifact ADD CONSTRAINT fk_artifact_buildrecord FOREIGN KEY (fk_artifact_buildrecord) REFERENCES buildrecord(id);
+
+    -- Copy data from old table
+    UPDATE artifact
+        SET buildrecord_id = build_record_built_artifact_map.build_record_id
+        FROM build_record_built_artifact_map
+        WHERE artifact.id = build_record_built_artifact_map.built_artifact_id;
+
+    -- Drop the old table
+    DROP TABLE build_record_built_artifact_map;
```
Comment (michalovjan) [inline model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql:219]: Are old associated indexes and constraints also dropped or do you have to do it manually?

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql (hunk 1 above)
Comment (janinko) [reply model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql:202]: Fixes `ERROR:  subquery in FROM must have an alias`

File: model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql (hunk 4 above)
Comment (janinko) [reply model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql:219]: Seems so.

**Review by jbartece — APPROVED** [review]
Looks OK, but I'm not an SQL interpret so, please try it :-)

File: model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql (hunk 1 above)
Comment (michalovjan) [reply model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql:202]: Sure!

File: model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql (hunk 4 above)
Comment (michalovjan) [reply model/src/main/resources/db-update/20000-from-1.8.x-to-2.0.0.sql:219]: ok

**Review by michalovjan — APPROVED** [review]

**Pull request merged**